src/Platform/Microsoft.Testing.Platform/Hosts/CommonTestHost.cs

```diff
@@ -61,6 +92,25 @@ public async Task<int> RunAsync()
         return exitCode;
     }
 
+    private async Task<bool> IsDotnetTestHandshakeSuccessfulAsync(NamedPipeClient dotnetTestPipeClient, ITestApplicationModuleInfo? testApplicationModuleInfo, CancellationToken testApplicationCancellationToken)
```

Comment:
When reading this method name, I would not expect it to actually do the handshake. Maybe `DoHandshakeAsync`? Having the `bool` return is indicating if that's successful or not.